src/Cmd/Add.cc

```diff
@@ -65,27 +67,51 @@ handleDependency(
 
 static std::string
 getDependencyGitUrl(const std::string_view dep) {
-  if (dep.find("://") == std::string_view::npos) {
-    // check if atleast in "user/repo" format
-    if (dep.find('/') == std::string_view::npos) {
-      logger::error("Invalid dependency: " + std::string(dep));
-      return "";
-    }
+  std::regex user_repo_regex(R"(^([^/]+)/([^/]+)$)");
+  std::regex url_regex(
+      R"((?:https?://)?(?:www\.)?(?:github|gitlab)\.com/([^/]+/[^/]+)(?:\.git)?)"
+  );
```

Comment:
Thank you for your contribution! For consistency and performance reasons, I generally avoid regex in this codebase, as custom parsers tend to offer better performance and readability. Could you please replace the regex with a custom parser here?

Reply:
Existing implementation works fine in that case, reverted the commit.